src/sentinel.c

```diff
@@ -4058,39 +4058,45 @@ NULL
                           (char*)c->argv[1]->ptr);
 }
 
-#define info_section_from_redis(section_name) do { \
-    if (defsections || allsections || !strcasecmp(section,section_name)) { \
-        sds redissection; \
-        if (sections++) info = sdscat(info,"\r\n"); \
-        redissection = genRedisInfoString(section_name); \
-        info = sdscatlen(info,redissection,sdslen(redissection)); \
-        sdsfree(redissection); \
-    } \
-} while(0)
 
 /* SENTINEL INFO [section] */
 void sentinelInfoCommand(client *c) {
+    
+    char defSections[11][15] = {"server", "clients", "memory", "persistence", "stats", "replication", "cpu", "modules", "errorstats", "cluster", "keyspace"};
```

Comment:
i don't like to repeat that list, or the logic. maybe we can find a better way?

disclaimer: i don't know that sentinel info command that well.
what does it do?
just forwards the call to redis's INFO and adds a `sentinel` section?
what's special of the 4 sections mentioned here (server, clients, cpu, stats)?
are these the only sections supported by SENTINEL info?

if i understand it correctly, i wanna propose this:
1. extract the code that generates the new dict and two booleans in server.c to a new function, one that will be used by both infoCommand and sentinelInfoCommand.
2. in sentinel, after calling it, purge any unknowns section names.
3. if the `all` or `everything` flags are set, just add the 4 default sections (server, clients, cpu, stats)
4. call genInfoString and avoid passing the 2 booleans.

will that do the trick?

Reply:
Hey oran, your understanding is right. I have extracted the code that generates the two dicts from server and sentinel and moved it to genInfoString, it should be a cleaner code now. Please take a look. Thank you

Reply:
I think you misunderstood me, or maybe i wasn't clear or my instructions were not valid.

I didn't mean that the logic that creates the dict should be moved from `infoCommand` into `genRedisInfoString`, and everyone just calls it.
what i meant was to extract that code to a new function, one that returns a dict.
then `infoCommand` calls that new function and passes it to `genRedisInfoString` and `sentinelInfoCommand` calls that new function to, then sanitizes it's output before calling `genRedisInfoString`,

i think `genRedisInfoString` should still take a dict and two booleans.
this will avoid the code duplication you had in your previous commit (parsing user request), and still keep the logic of parsing the user request and generating the output string separate.

is there anything wrong it that approach which lead you to what you did?

Reply:
I understand what you're saying, I think what you're saying is good. I'll change it to that. The reason for adding the logic to `genRedisInfoString` was that with that approach the default sections array then would not be needed and just default could've been added to the dictionary.
Also for the if statement at the end of `genRedisInfoString` which generates the info for the modules sections, should we assume that there was only one argument for the infocommand for example "info mudulesection" or do we want it for multiple sections too?

Reply:
i think we want it to support multiple sections.
so `RedisModuleInfoCtx` will contain that dict rather than just one string, and `RM_InfoAddSection` will search that dict instead of doing string compare.